packages/vx-text/src/Text.js

```diff
@@ -9,13 +9,11 @@ class Text extends Component {
     this.state = {
       wordsByLines: []
     };
-  }
 
-  componentWillMount() {
     this.updateWordsByLines(this.props, true);
   }
 
-  componentWillReceiveProps(nextProps) {
+  static getDerivedStateFromProps(nextProps) {
     const needCalculate =
       this.props.children !== nextProps.children || this.props.style !== nextProps.style;
```

Comment:
The change is unfortunately not that easy. `getDerivedStateFromProps` doesn't have access to the component instance, which means no `this`. As a second, `getDerivedStateFromProps` needs to return a state object or null if nothing changes, much like `setState` because it runs right before `render`.

More info can be found in the React Docs.  
https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops